src/arrayMethodJoin.js

```diff
@@ -4,8 +4,22 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i > 0) {
+        string = string + separator;
+      }
+
+      if (this[i] === null || this[i] === undefined) {
+        continue;
+      } else {
+        string = string + this[i];
+      }
+    }
```

Comment:
the use of an if...else statement with a continue statement can be simplified. You can use an if statement without an else block to only execute the code if the condition is true

Reply:
done